Can't build with gcc 5 where __AVX512F__ is defined

Seems like the `_mm512_*` functions used in the `#ifdef USE_AVX512` block are only present when using gcc 7+. on 5 it fails with errors like 

``` src/annoylib.h: In function ‘T {anonymous}::euclidean_distance(const T*, const T*, int) [with T = float]’:
src/annoylib.h:304:36: error: ‘_mm512_reduce_add_ps’ was not declared in this scope
result = _mm512_reduce_add_ps(d);```

--- a/setup.py
+++ b/setup.py
@@ -39,10 +39,6 @@ with codecs.open('README.rst', encoding='utf-8') as fobj:
 extra_compile_args = []
 extra_link_args = []
 
-if os.environ.get('TRAVIS') == 'true':
-    # Resolving some annoying issue
-    extra_compile_args += ['-mno-avx']
-
 # Not all CPUs have march as a tuning parameter
 cputune = ['-march=native',]
 if platform.machine() == 'ppc64le':

--- a/src/annoylib.h
+++ b/src/annoylib.h
@@ -75,9 +75,11 @@ typedef unsigned __int64  uint64_t;
 #endif
 
 #ifndef NO_MANUAL_VECTORIZATION
-#if defined(__AVX512F__)
+#if defined(__GNUC__) && (__GNUC__ >6) && defined(__AVX512F__)  // See #402
+#pragma message "Using 512-bit AVX instructions"
 #define USE_AVX512
 #elif defined(__AVX__) && defined (__SSE__) && defined(__SSE2__) && defined(__SSE3__)
+#pragma message "Using 128-bit AVX instructions"
 #define USE_AVX
 #endif
 #endif
@@ -988,6 +990,8 @@ public:
     } else {
       // Delete file if it already exists (See issue #335)
       unlink(filename);
+
+      printf("path: %s\n", filename);
 
       FILE *f = fopen(filename, "wb");
       if (f == NULL) {

--- a/src/annoylib.h
+++ b/src/annoylib.h
@@ -74,14 +74,14 @@ typedef unsigned __int64  uint64_t;
 #define popcount cole_popcount
 #endif
 
-#ifndef NO_MANUAL_VECTORIZATION
-#if defined(__GNUC__) && (__GNUC__ >6) && defined(__AVX512F__)  // See #402
+#if !defined(NO_MANUAL_VECTORIZATION) && defined(__GNUC__) && (__GNUC__ >6) && defined(__AVX512F__)  // See #402
 #pragma message "Using 512-bit AVX instructions"
 #define USE_AVX512
-#elif defined(__AVX__) && defined (__SSE__) && defined(__SSE2__) && defined(__SSE3__)
+#elif !defined(NO_MANUAL_VECTORIZATION) && defined(__AVX__) && defined (__SSE__) && defined(__SSE2__) && defined(__SSE3__)
 #pragma message "Using 128-bit AVX instructions"
 #define USE_AVX
-#endif
+#else
+#pragma message "Using no AVX instructions"
 #endif
 
 #if defined(USE_AVX) || defined(USE_AVX512)

